numba/cpython/cmathimpl.py

```diff
@@ -183,21 +180,31 @@ def log10_impl(z):
         # See http://bugs.python.org/issue22544
         return complex(z.real / LN_10, z.imag / LN_10)
 
-    res = context.compile_internal(builder, log10_impl, sig, args)
-    return impl_ret_untracked(context, builder, sig, res)
+    return log10_impl
 
 
-@lower(cmath.phase, types.Complex)
-@intrinsic_complex_unary
-def phase_impl(x, y, x_is_finite, y_is_finite):
+@overload(cmath.phase)
+def phase_impl(x):
     """cmath.phase(x + y j)"""
-    return math.atan2(y, x)
 
-@lower(cmath.polar, types.Complex)
-@intrinsic_complex_unary
-def polar_impl(x, y, x_is_finite, y_is_finite):
-    """cmath.polar(x + y j)"""
-    return math.hypot(x, y), math.atan2(y, x)
+    if not isinstance(x, types.Complex):
+        return
+
+    def impl(x):
+        r, i = x.real, x.imag
+        return math.atan2(i, r)
+    return impl
+
+
+@overload(cmath.polar)
+def polar_impl(x):
+    if not isinstance(x, types.Complex):
+        return
+
+    def impl(x):
+        r, i = x.real, x.imag
```

Comment:
Just confirming, previously these were assigned using the `intrinsic_complex_unary` decorator, and now it's being done during compile time.

Reply:
This is fine. `intrinsic_complex_unary` unpacks the complex value and call the decorated function with both `real` and `imag` values as well as if they're finite or not.